### PR TITLE
[Sprint: 51] XD-3115 Support Reactor Processors that have an inheritance hierarchy

### PR DESCRIPTION
**Pull request by markpollack**

### CODE REVIEW

File: spring-xd-reactor/src/main/java/org/springframework/xd/reactor/ReactorReflectionUtils.java [hunk 1]
```diff
@@ -36,22 +38,29 @@
      * be determined.
```
Comment (ericbottard) [inline spring-xd-reactor/src/main/java/org/springframework/xd/reactor/ReactorReflectionUtils.java]: 2 lines above (can't comment), there is a type. Should read `its`, not `it's`. Also, terminate sentence with a dot.

File: spring-xd-reactor/src/main/java/org/springframework/xd/reactor/ReactorReflectionUtils.java [hunk 2]
```diff
@@ -36,22 +38,29 @@
...
-
-            t = pt.getActualTypeArguments()[0];
-            if (t instanceof ParameterizedType) {
-                return (Class) ((ParameterizedType) t).getRawType();
-            } else if (t instanceof Class) {
-                return (Class) t;
+        Class<?> searchType = processor.getClass();
+        while (searchType != Object.class) {
+            if (searchType.getClass().getGenericInterfaces().length == 0) {
+                continue;
```
Comment (cppwfs) [inline spring-xd-reactor/src/main/java/org/springframework/xd/reactor/ReactorReflectionUtils.java]: Wouldn't this be an infinite loop?

**Comment by markpollack** [comment]
Need to rewrite this from scratch...will reissue new PR

**Comment by mbogoevici** [comment]
@markpollack actually rewrote this here: https://github.com/mbogoevici/spring-xd/commit/ab2d4da447321726af621bb2661d3995b4b5caea

can make it a new PR

**Comment by markpollack** [comment]
Closing, will have a new PR.

**Pull request closed without merging**